plugins/mac-avcapture/OBSAVCapture.m

```diff
@@ -700,21 +688,66 @@ + (NSString *)stringFromSettings:(void *)settings withSetting:(NSString *)settin
     return result;
 }
 
++ (NSString *)effectsWarningForDevice:(AVCaptureDevice *)device
+{
+    int effectsCount = 0;
+    NSString *effectWarning = nil;
+    if (@available(macOS 12.0, *)) {
+        if (device.portraitEffectActive) {
+            effectWarning = @"Warning.Effect.Portrait";
+            effectsCount++;
+        }
+    }
+    if (@available(macOS 12.3, *)) {
+        if (device.centerStageActive) {
+            effectWarning = @"Warning.Effect.CenterStage";
+            effectsCount++;
+        }
+    }
+    if (@available(macOS 13.0, *)) {
+        if (device.studioLightActive) {
+            effectWarning = @"Warning.Effect.StudioLight";
+            effectsCount++;
+        }
+    }
+    if (@available(macOS 14.0, *)) {
+        /// Reaction effects do not follow the same paradigm as other effects in terms of checking whether they are active. According to Apple, this is because a device instance property `reactionEffectsActive` would have been ambiguous (conflicting with whether a reaction is currently rendering). Instead, Apple exposes the `AVCaptureDevice.reactionEffectGesturesEnabled` class property (which apparently exists, but is hidden/private for all other effects) to tell us whether the effect is enabled application-wide, while `device.canPerformReactionEffects` tells us whether the device's active format currently supports the effect. The logical conjunction of these two properties tells us whether putting our thumbs inside the video frame will make fireworks appear. The device instance properties for other effects are a convenience 'shorthand' for this private class/instance property combination.
```

Comment:
Please add some paragraph breaks (you should be able to add an empty line via `///`) to make this more readable.

Reply:
Changed. Used one line of code per paragraph still; let me know if you want different formatting (such as picking a column width and justifying manually with more lines, etc.)